### PR TITLE
Serena Roderick

### PR DESCRIPTION
**Pull request by serenaroderick-techconsulting**
added view model

### CODE REVIEW

File: Weather/WeatherData.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//  Weather
+//
+//  Created by Serena Roderick on 9/13/24.
+//
+
+import Foundation
+
+import Foundation
+
+// MARK: - Welcome
```
Comment (codemanager255) [inline Weather/WeatherData.swift:12]: what is welcome here , use proper naming

File: Weather/WeatherData.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+struct WeatherData: Codable {
```
Comment (codemanager255) [inline Weather/WeatherData.swift:13]: use decodable instead of codable

File: Weather/WeatherViewController.swift [hunk 3]
```diff
@@ -7,7 +7,8 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class WeatherViewController: UIViewController {
```
Comment (codemanager255) [inline Weather/WeatherViewController.swift:10]: delete this file if not used

File: Weather/WeatherViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    func handleApiData()
+    func passDataToView(data: Data)
+    func kelvinToCelcius(kelvin: Double) -> Double
+    func kelvinToFarenheit(kelvin: Double) -> Double
+}
+
+class WeatherViewModel: WeatherViewModelProtocol {
+
+    var weatherData: [WeatherData]?
+    var networkManager: NetworkManager?
```
Comment (codemanager255) [inline Weather/WeatherViewModel.swift]: is this mutlabe, can we use let ,    insert this as mandtory depdency using intilizer

Comment (codemanager255) [inline Weather/WeatherViewModel.swift]: can it be private ?

File: Weather/WeatherViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    
+    func handleApiData() {
+        WeatherViewController.weatherData = weatherData
+    }
+    
+    func passDataToView(data: Data) {
+        
+    }
+    
+    func kelvinToFarenheit(kelvin: Double) -> Double {
```
Comment (codemanager255) [inline Weather/WeatherViewModel.swift:43]: unit test cases for this

**Review by jpickett-techconsulting — APPROVED** [review]
Yass

**Pull request merged**